src/java/org/apache/cassandra/io/IVersionedSerializer.java

```diff
@@ -17,6 +17,34 @@
  */
 package org.apache.cassandra.io;
 
+import java.io.IOException;
+
+import org.apache.cassandra.io.util.DataInputPlus;
+import org.apache.cassandra.io.util.DataOutputPlus;
+
 public interface IVersionedSerializer<T> extends IVersionedAsymmetricSerializer<T, T>
 {
+    static <T> IVersionedSerializer<T> fromSerializer(Serializer<T> delegate)
```

Comment:
TODO: this API isn't consistent, should look closer to make sure the APIs match for each interface